src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -66,12 +66,12 @@ export class SelectBox extends vsSelectBox {
 
 	constructor(options: SelectOptionItemSQL[] | string[], selectedOption: string, contextViewProvider: IContextViewProvider, container?: HTMLElement, selectBoxOptions?: ISelectBoxOptions) {
 		let optionItems: SelectOptionItemSQL[];
-		if (Array.isArray<string>(options)) {
-			optionItems = (options as string[]).map(o => {
+		if (typeof (options[0]) === 'string' && Array.isArray<string>(options)) {
```

Comment:
Huh, so isArray<string> isn't actually checking that it's an array of strings? That seems like a bug there?

Reply:
No it's not a bug. Remember that the type is only there during compile time, not run time.

Reply:
If that was the case then typeof(options[0]) wouldn't do anything 😄 Interfaces are dropped (so yeah we lose the typing for SelectOptionItemSQL) but the string type is still there so theoretically we should be able to actually make a generic version of isArray unless I'm missing something that at least is able to handle classes and built in types correctly.

I think the problem here is that we define the functions but they're never actually implemented anywhere so it just falls back to using the built-in isArray. Looking at VS Code it's likely they added these purely for the type coercion and didn't actually care about differentiating different types of arrays (since all their uses are like `string | string[]`